presto-main-base/src/test/java/com/facebook/presto/sql/query/TestColumnMask.java

```diff
@@ -0,0 +1,382 @@
+            accessControl.reset();
+            accessControl.columnMask(
+                    new QualifiedObjectName(CATALOG, "tiny", "orders"),
+                    "orderkey",
+                    USER,
+                    new ViewExpression(RUN_AS_USER, Optional.of(CATALOG), Optional.of("tiny"), "count(*) > 0"));
+
+            assertions.assertFails("SELECT orderkey FROM orders", "\\Qline 1:10: Column mask for 'orders.orderkey' cannot contain aggregations, window functions or grouping operations: [\"count\"(*)]\\E");
+        });
+
+        // window function
+        assertions.executeExclusively(() -> {
+            accessControl.reset();
+            accessControl.columnMask(
+                    new QualifiedObjectName(CATALOG, "tiny", "orders"),
+                    "orderkey",
+                    USER,
+                    new ViewExpression(RUN_AS_USER, Optional.of(CATALOG), Optional.of("tiny"), "row_number() OVER () > 0"));
+
+            assertions.assertFails("SELECT orderkey FROM orders", "\\Qline 1:22: Column mask for 'orders.orderkey' cannot contain aggregations, window functions or grouping operations: [\"row_number\"() OVER ()]\\E");
+        });
+
+        // grouping function
+        assertions.executeExclusively(() -> {
+            accessControl.reset();
+            accessControl.columnMask(
+                    new QualifiedObjectName(CATALOG, "tiny", "orders"),
+                    "orderkey",
+                    USER,
+                    new ViewExpression(USER, Optional.of(CATALOG), Optional.of("tiny"), "grouping(orderkey) = 0"));
+
+            assertions.assertFails("SELECT orderkey FROM orders", "\\Qline 1:20: Column mask for 'orders.orderkey' cannot contain aggregations, window functions or grouping operations: [GROUPING (orderkey)]\\E");
+        });
+    }
+
+    @Test
+    public void testInsertWithColumnMasking()
+    {
+        assertions.executeExclusively(() -> {
+            accessControl.reset();
+            accessControl.columnMask(
+                    new QualifiedObjectName(CATALOG, "tiny", "orders"),
+                    "clerk",
+                    USER,
+                    new ViewExpression(USER, Optional.of(CATALOG), Optional.of("tiny"), "clerk"));
+
+            assertions.assertFails("INSERT INTO orders SELECT * FROM orders", "Insert into table with column masks is not supported");
+        });
+    }
+
+    @Test
+    public void testDeleteWithColumnMasking()
+    {
+        assertions.executeExclusively(() -> {
+            accessControl.reset();
+            accessControl.columnMask(
+                    new QualifiedObjectName(CATALOG, "tiny", "orders"),
+                    "clerk",
+                    USER,
+                    new ViewExpression(USER, Optional.of(CATALOG), Optional.of("tiny"), "clerk"));
+
+            assertions.assertFails("DELETE FROM orders", "\\Qline 1:1: Delete from table with column mask is not supported\\E");
+        });
+    }
+}
```

Comment:
Can we also do a few tests with complex/nested types such as ROW and ARRAY types

Reply:
I suppose it would be good to make sure a `ViewExpression` can apply to a complex type. It would take me some time to work this out, would that be ok to work on as a followup?

Reply:
Can you at least open a ticket and link it to this discussion so we don't lose track of this?

Reply:
Issue created: https://github.com/prestodb/presto/issues/24998